File: live-test/java/src/main/java/demo/Probes.java
// Scope-discrimination fixture, production side: Probe.target() has callers
// and a subtype in production sources here, with counterparts in the test
// source root, so scoped queries split results by production, test, or class.
package demo;

class Probe {
    int target() {
        return 42;
    }

    int sameClassCaller() {
        return target() + 1;
    }
}

// A caller outside Probe, separating same-class callers from the rest of
// production.
class ProbeAux {
    static int freeProdCaller() {
        return new Probe().target();
    }
}

// Production-side subtype; its counterpart lives in the test source root.
class ProbeProdChild extends Probe {
}

File: live-test/kotlin/src/main/kotlin/demo/Probes.kt
// Production side of the scope-discrimination pair: callers of target() and
// subtypes of Probe are probed under production, test, this-class and
// this-module scopes; the test-only hits live in the test source tree.
package demo

open class Probe {
    fun target(): Int = 42

    // Same-class caller: what a this-class-scoped caller search can see.
    fun sameClassCaller(): Int = target() + 1
}

// Top-level production caller: outside the class, inside production sources.
fun freeProdCaller(): Int = Probe().target()

// Production-side subtype; the test tree adds another so subtype results can
// differ by scope.
class ProbeProdChild : Probe()
